src/main/java/command/AutoClearCommand.java

```diff
@@ -0,0 +1,53 @@
+package command;
+
+import event.Event;
+import eventlist.EventList;
+import exception.EmptyEventListException;
+import exception.NoDateBeforeException;
+import exception.NoEventOneMonthAgoException;
+import exception.NuScheduleException;
+import locationlist.BusStopList;
+import locationlist.LocationList;
+import storage.Storage;
+import ui.UI;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+
+/**
+ * Represents the command call when the user wants to turn on/off the auto-clear function
+ * everytime the user input the function, events of one month ago will be cleared
+ */
+public class AutoClearCommand extends Command {
+
+    protected LocalDate clearDate;
+
+    public AutoClearCommand() {
+        clearDate = LocalDate.now().minusMonths(1);
+    }
+
+    /**
+     * Execute the command based on the specific command type.
+     *
+     * @param events    the list of Events.
+     * @param locations the list of Locations.
+     * @param busStops  the list of BusStops.
+     * @param ui        do outputs.
+     * @param storage   store the data.
+     * @throws EmptyEventListException the exceptions when the user try to clear an empty list.
+     */
+    @Override
+    public void execute(EventList events, LocationList locations, BusStopList busStops, UI ui, Storage storage) throws EmptyEventListException, NoEventOneMonthAgoException {
+        if (events != null) {
+            ArrayList<Event> filteredEventList = events.filterDateBefore(clearDate);
+            if (filteredEventList.size() == 0) {
+                throw new NoEventOneMonthAgoException();
+            } else {
+                ui.printAutoClearOn();
+                events.clearBefore(clearDate);
```

Comment:
Auto Clear in our product should be considered as an option that can be opened/closed by the user, so maybe this Command does not behave as what we want it to be? Perhaps you would like to consider this as a flag to be set, which will be checked every time when launching this app.